src/field-modules/my-logs/components/EditLog.vue

```diff
@@ -662,12 +667,12 @@ export default {
 
     addQuant() {
       const quanTemplate = {
-        measure: 'weight',
-        value: 0,
-        unit: {id: 0, resource: 'taxonomy_term'},
-        label: '',
+        measure: null,
+        value: null,
+        unit: null,
+        label: null,
       };
-      this.logs[this.currentLogIndex].quantity.data.push(quanTemplate);
+      this.logs[this.currentLogIndex].quantity.data.unshift(quanTemplate);
```

Comment:
We should not be mutating an array in place like this, especially if it's in the store (I'm actually surprised that Vuex allows us to do this). Instead we should use the `updateLog` store mutation to modify the contents of the store.

I would suggest getting rid of `addQuant` altogether, because it seems like some unnecessary indirection; `updateNewQuant` can conditionally commit the `updateLog` mutation with the `quantTemplate` as payload, if it's a new quantity, rather than calling `addQuant`. Just make sure it's not being called elsewhere.

Reply:
I see what you mean about directing new quantities through the store, rather than mutating the quantity object directly.  I can definitely do that.  And I agree that we might as well get rid of addQuant as a separate function.

Reply:
Still working on this one - sorry for the slow progress!  I'll have the fix done by tomorrow morning.  

Just so you know, I may want to retain the addQuant() function, as I'm no longer convinced that this is unnecessary indirection.  Update New Quant is used to update the existing quantity, and it calls addQuant if there is no existing quantity.  addQuant is also called when the user hits the "add another quantity" button.  To me, it makes sense to separaet the tasks of creating a new quantity and updating an existing quantity into their own functions.

Reply:
No rush, @alexadamsmith. I'm just packaging a release right now, so these updates can wait til the next release, which probably won't be until I fix the camera issues.

I would still prefer we eliminate `addQuant` in the interest of keeping the code DRY and easier to reason about. If we keep it, we'll need to amend it to call the store and commit an `updateLog` mutation, which is needless repetition of what `updateNewQuant` is already doing. By creating another call to `updateLog`, it's a potential another place you have to look when trying to determine where in the code that mutation originated. It also means that, to understand the full logic of `updateNewQuant`, you need to hunt around in a different section of the code to find `addQuant` (this is what I meant by indirection). It's better to have all the code related to updating a quantity in one place, whether it's a brand new quantity, or an existing one. It's not overloading the responsibility of `updateNewQuant`, in my opinion, to cover both of those cases, precisely because they are so similar, a fact borne out by the need for `addQuant` to duplicate of the behavior of `updateNewQuant` if we keep it separate.

Reply:
Just FYI, I got the all mechanics ticking along and using updateLog mutations, following the same pattern as the rest of EditLog.  updateNewQuant and addQuant both pass their data to the updateCurrentLog function.  There the data are packaged with a timestamp and other props, then sent along to updateLog. 

It shouldn't be too difficult to combine updateNewLog and addQuant.  I can just add a new prop to updateNewLog that allows the function to determine whether it is being called by the 'new log' button, or by data being entered in a field.

Since there's not a rush, I am going to take a bit of time this evening to double check my work and combine those functions into one.  

Thanks for all the feedback!

Reply:
Sounds great! Thanks for putting in the extra effort! :clap: